AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -138,60 +100,73 @@ class CompleteScreenViewModel: ObservableObject {
         exitRoute()
     }
     
+    func confirmationButtonPressed() {
+        following()
+        saveToDb()
+    }
+    
+    private func following() {
+        // This doesn't work and I don't know why
+        completeButtonEnabled = false
+        completeButtonText = "Following..."
+    }
+    
+    private func saveToDb() {
+        guard let externalSessionWithStreams = externalSessionWithStreams else {
+            assertionFailure("Confirmation button pressed when there was no session with streams")
+            return
+        }
+        
+        do {
+            try externalSessionsStore.createExternalSession(session: externalSessionWithStreams)
```

Comment:
since we have a `Controller` in place, the VM should not really handle storage/db/other services directly